subsys/bluetooth/mesh/provisioner.c

```diff
@@ -0,0 +1,767 @@
+}
+
+static void prov_random(const uint8_t *data)
+{
+	uint8_t conf_verify[16];
+
+	BT_DBG("Remote Random: %s", bt_hex(data, 16));
+	if (!memcmp(data, bt_mesh_prov_link.rand, 16)) {
+		BT_ERR("Random value is identical to ours, rejecting.");
+		prov_fail(PROV_ERR_CFM_FAILED);
+		return;
+	}
+
+	if (bt_mesh_prov_conf(bt_mesh_prov_link.conf_key,
+			      data, bt_mesh_prov_link.auth, conf_verify)) {
+		BT_ERR("Unable to calculate confirmation verification");
+		prov_fail(PROV_ERR_UNEXP_ERR);
+		return;
+	}
+
+	if (memcmp(conf_verify, bt_mesh_prov_link.conf, 16)) {
+		BT_ERR("Invalid confirmation value");
+		BT_DBG("Received:   %s", bt_hex(bt_mesh_prov_link.conf, 16));
+		BT_DBG("Calculated: %s",  bt_hex(conf_verify, 16));
+		prov_fail(PROV_ERR_CFM_FAILED);
+		return;
+	}
+
+	if (bt_mesh_prov_salt(bt_mesh_prov_link.conf_salt,
+			      bt_mesh_prov_link.rand, data, bt_mesh_prov_link.prov_salt)) {
+		BT_ERR("Failed to generate provisioning salt");
+		prov_fail(PROV_ERR_UNEXP_ERR);
+		return;
+	}
+
+	BT_DBG("ProvisioningSalt: %s", bt_hex(bt_mesh_prov_link.prov_salt, 16));
+
+	send_prov_data();
+}
+
+static void prov_confirm(const uint8_t *data)
+{
+	BT_DBG("Remote Confirm: %s", bt_hex(data, 16));
+
+	memcpy(bt_mesh_prov_link.conf, data, 16);
+
+	send_random();
+}
+
+static void prov_failed(const uint8_t *data)
+{
+	BT_WARN("Error: 0x%02x", data[0]);
+	reset_state();
+}
+
+static void local_input_complete(void)
+{
+	if (atomic_test_and_clear_bit(bt_mesh_prov_link.flags, WAIT_CONFIRM)) {
+		send_confirm();
+	}
+}
+
+static void prov_link_closed(void)
+{
+	reset_state();
+}
+
+static void prov_link_opened(void)
+{
+	send_invite();
+}
+
+static const struct bt_mesh_prov_role role_provisioner = {
+	.input_complete = local_input_complete,
+	.link_opened = prov_link_opened,
+	.link_closed = prov_link_closed,
+	.error = prov_fail,
+	.op = {
+		[PROV_CAPABILITIES] = prov_capabilities,
+		[PROV_PUB_KEY] = prov_pub_key,
+		[PROV_INPUT_COMPLETE] = prov_input_complete,
+		[PROV_CONFIRM] = prov_confirm,
+		[PROV_RANDOM] = prov_random,
+		[PROV_COMPLETE] = prov_complete,
+		[PROV_FAILED] = prov_failed,
+	},
+};
+
+static void prov_set_method(uint8_t method, uint8_t action, uint8_t size)
+{
+	bt_mesh_prov_link.oob_method = method;
+	bt_mesh_prov_link.oob_action = action;
+	bt_mesh_prov_link.oob_size = size;
+}
+
+int bt_mesh_auth_method_set_input(bt_mesh_input_action_t action, uint8_t size)
+{
+	if (!action || !size) {
+		return -EINVAL;
+	}
+
+	prov_set_method(AUTH_METHOD_INPUT, find_msb_set(action) - 1, size);
+	return 0;
+}
+
+int bt_mesh_auth_method_set_output(bt_mesh_output_action_t action, uint8_t size)
+{
+	if (!action || !size || size > 8) {
+		return -EINVAL;
+	}
+
+	prov_set_method(AUTH_METHOD_OUTPUT, find_msb_set(action) - 1, size);
+	return 0;
+}
+
+int bt_mesh_auth_method_set_static(const uint8_t *static_val, uint8_t size)
+{
+	if (!size || !static_val || size > 16) {
+		return -EINVAL;
+	}
+
+	prov_set_method(AUTH_METHOD_STATIC, 0, size);
```

Comment:
```suggestion
	prov_set_method(AUTH_METHOD_STATIC, 0, 0);
```
so we don't end up sending non-zero by accident after refactoring.